internal/controller/template_controller.go

```diff
@@ -419,14 +412,18 @@ func processTemplateResources(
 		log.Error(err, "unable to create template from resource")
 		return err
 	}
+
 	o, err := t.Evaluate(re)
 	if err != nil {
 		log.Error(err, "unable to evaluate template")
 		return err
 	}
 
-	fmt.Println(o.Output)
+	// Prepare to track desired resources
+	desiredResources := make(map[string]*unstructured.Unstructured)
+	desiredResourceIdentifiers := make([]ResourceIdentifier, 0)
```

Comment:
I _think_ the code will end up being nicer if you use a set type for this and the existing resources, and then do a set difference to find the ones to delete.

Go doesn’t have a built-in set type. A quick google search showed:
* [k8s.io/apimachinery/pkg/util/sets](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set) — might be the best option as I think using it won’t add another dependency to `go.mod`
* [github.com/deckarep/golang-set/v2](https://pkg.go.dev/github.com/deckarep/golang-set/v2) — has been around for a while and looks lightweight

Reply:
Done. Used the second package since the first one only accepts limited types (string, int, )
The Second one accepts any comparable type